src/scimlfunctions.jl

```diff
@@ -424,7 +419,7 @@ struct ODEFunction{iip, specialize, F, TMM, Ta, Tt, TJ, JVP, VJP, JP, SP, TW, TW
     colorvec::TCV
     sys::SYS
     initialization_data::ID
-    nlprob::NLP
+    nlprob_data::NLP
```

Comment:
We should probably type restrict both of these?

Reply:
yeah, I was copying `initialization_data`. Should we restrict both?

Reply:
yes